apps/src/types/progressTypes.ts

```diff
@@ -0,0 +1,140 @@
+// Typescript definitions for types relating to progress. Note that many
+// of these are "duplicated" by `/templates/progress/progressTypes, which defined
+// these using PropTypes for usage in Javascript React components. As we move towards
+// typescript we can deprecate the PropTypes definitions and use these instead.
+
+/**
+ * See ApplicationHelper::PUZZLE_PAGE_NONE.
+ */
+export const PUZZLE_PAGE_NONE = -1;
+
+export interface Lesson {
```

Comment:
I assume these fields are copied from what is already on the page? How easy is it to change these in the future (asking as a complete TypeScript newbie)

Reply:
Yep these are either based on what we are already storing, or if they were defined in [progressTypes.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/progress/progressTypes.js) (which is a prop types definition file), I just copied those. It's very easy to change these in the future, and the benefit of typescript is if you are using the type somewhere and you update the type, if your old code no longer works you will get a compiler error :) So it will catch errors before running.